components/encryption/src/master_key/kms.rs

```diff
@@ -256,19 +308,31 @@ impl KmsBackend {
             other => {
                 return Err(box_err!(
                     "KMS vendor mismatch expect {:?} got {:?}",
-                    AWS_KMS_VENDOR_NAME,
+                    vendor_name,
                     other
                 ))
             }
         }
 
-        let mut inner = self.inner.lock().unwrap();
-        let ciphertext_key = content.metadata.get(MetadataKey::KmsCiphertextKey.as_str());
-        if ciphertext_key.is_none() {
-            return Err(box_err!("KMS ciphertext key not found"));
+        let ciphertext_key = match content.metadata.get(MetadataKey::KmsCiphertextKey.as_str()) {
+            None => return Err(box_err!("KMS ciphertext key not found")),
+            Some(key) => Encrypted::new(key.to_owned())?,
+        };
+
+        let mut state = self.state.lock().unwrap();
+        self.lazy_load_state(&mut *state)?;
```

Comment:
Add comment to say the KMS data key is rotated every time on restart? This is different from previous behavior but I think it is a good change.

Reply:
okay, the prior behavior is restored now